website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,547 @@
+    {%- if  env_var('DBT_ENV_TYPE','DEV') == 'DEV' -%}
+    
+        {#- we replace characters not allowed in the schema names by "_" -#}
+        {%- set re = modules.re -%}
+        {%- set cleaned_branch = re.sub("\W", "_", env_var('DBT_CLOUD_GIT_BRANCH')) -%}
+        
+        {%- if custom_schema_name is none -%}
+
+            {{ cleaned_branch }}
+
+        {%- else -%}
+
+             {{ cleaned_branch }}_{{ custom_schema_name | trim }}
+
+        {%- endif -%}
+        
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+</File>
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Branch      |Target database| Target schema | Resulting object                  |
+|-------------|:----------:|:-------------:|:-------------:|:---------------------------------:|
+| Developer 1 |`featureABC`|dev            | dbt_dev1      |dev.featureABC_marketing.my_model  |
+| Developer 2 |`featureABC`|dev            | dbt_dev2      |dev.featureABC_marketing.my_model  |
+| Developer 1 |`feature123`|dev            | dbt_dev1      |dev.feature123_marketing.my_model  |
+| CI PR 123   |            |ci             | dbt_pr_123    |ci.dbt_pr_123_marketing.my_model   |
+| CI PR 234   |            |ci             | dbt_pr_234    |ci.dbt_pr_234_marketing.my_model   |
+| Production  |            |prod           | analytics     |prod.marketing.my_model\           |
+
+
+When developer 1 and developer 2 are checked out on the same branch, they will generate the same object in the data warehouse. This shouldn't be a problem as being on the same branch means the model's code will be the same for both developers.
+
+
+### 4. Use a static schema for CI 
+
+Some organizations prefer to write their CI jobs to a single schema with the PR identifier prefixed to the front of the table name. It's important to note that this will result in long table names. 
+
+To do so, you can create a new file called `generate_schema_name.sql` under your macros folder with the following code:
+
+
+<File name='macros/generate_schema_name.sql'>
+
+```jinja
+
+{% macro generate_schema_name(custom_schema_name=none, node=none) -%}
+
+    {%- set default_schema = target.schema -%}
+    
+    {# If the CI Job does not exist in its own environment, use the target.name variable inside the job instead #}
+    {# {%- if target.name == 'CI' -%} #} 
+    
+    {%- if env_var('DBT_ENV_TYPE','DEV') == 'CI' -%}
+        
+        ci_schema
+        
+    {%- elif custom_schema_name is none -%}
+        
+        {{ default_schema }}
+    
+    {%- else -%}
+        
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+    
+    {%- endif -%}    
+
+{%- endmacro %}
+
+```
+</File>
+
+
+<File name='macros/generate_schema_name.sql'>
```

Comment:
since this is the alias macro, shouldn't this be 'macros/generate_alias_name' and not 'generate_schema_name.sql'?

Reply:
Yep, right again!